go/vt/vtgate/planbuilder/physical/subquery_planning.go

```diff
@@ -82,6 +82,37 @@ func optimizeSubQuery(ctx *plancontext.PlanningContext, op *abstract.SubQuery) (
 	return outerOp, nil
 }
 
+// Remove the given subquery from any predicates and vindex predicates.
+//
+// If a subquery is merged into it's outer query, it needs to be removed from
+// the route as it can't be used as a routing predicate anymore.
+func RemoveExtractedSubqueryPredicate(route *Route, extractedSubquery *sqlparser.ExtractedSubquery) {
+	for _, vp := range route.VindexPreds {
+		idx := -1
+
+		for j, option := range vp.Options {
+			if sqlparser.EqualsExpr(option.Predicates[0], extractedSubquery.Original) {
```

Comment:
This only looks at the first predicate in the list. I think that's correct for all single column vindexes. Do we need to do anything special here for multi column vindexes? I don't think there's any other support for subqueries that match against tuples, so multi column vindex (and thus vindex predicates with more than one value) are not relevant yet.

Reply:
I think the safest bet would be to check the length of `option.Predicates`, and, only if the length is equal to one, compare the two predicates.